### PR TITLE
Update max_time.rst

### PR DESCRIPTION
**Pull request by renier**
Add other limitations to new limitations document.

Will help with frequent questions on slack on this subject.

This type of information used to be here https://github.com/Qiskit-Partners/qiskit-runtime/blob/main/docs/limitations.rst
But that repo has been archived.

### CODE REVIEW

File: docs/max_time.rst [hunk 1]
```diff
@@ -5,7 +5,8 @@ Maximum Execution Time
 ======================
 
 To ensure fairness, there is a maximum execution time for each Qiskit Runtime job. If
-a job exceeds this time limit, it is forcibly terminated. The maximum execution time is the
+a job exceeds this time limit, it is forcibly canceled. This is represented in the job
+status as `Canceled - Ran too long`. The maximum execution time is the
```
Comment (daka1510) [inline docs/max_time.rst:9]: With https://github.com/Qiskit/qiskit-ibm-runtime/issues/168 we will move away from the deprecated job status string and instead use the new `state` property

```
"state": {
    "status": "Cancelled",
    "reason": "Ran too long"
}
```

I assume this change will be transparent for users and we can keep it as-is here...

**Review by daka1510 — APPROVED** [review]
LGTM.

File: docs/max_time.rst (hunk 1 above)
Comment (renier) [reply docs/max_time.rst:9]: Yes, since the UI will continue to present it as one thing.

**Review by rathishcholarajan — CHANGES_REQUESTED** [review]
Looks good mostly, suggested minor changes…

**Review by rathishcholarajan — APPROVED** [review]

**Pull request merged**